src/main/java/org/apereo/openequella/tools/toolbox/checkFiles/CheckFilesDbHandler.java

```diff
@@ -283,11 +286,112 @@ private boolean doesAttachmentExist(ResultsRow attRow) {
+    logger.info("Att type [{}].", attRow.getAttType());
+
+    final File stdFile = new File(path);
+    final File altFile = new File(altPath);
 
-    if ((new File(path)).exists()) {
+    if (attRow.getAttType().equals(Config.ATT_TYPE_ZIP_ENTRY)) {
+      if (attRow.getAttStatus().equals("IN_PROGRESS_ZIP_DIR")) {
+        if ((stdFile.exists() && stdFile.isDirectory())
+            || (altFile.exists() && altFile.isDirectory())) {
+          return true;
+        }
+        return failAndRunChecks(stdFile, altFile, false);
+      }
+      // Otherwise, its supposed to be a file
+      if ((stdFile.exists() && stdFile.isFile()) || (altFile.exists() && altFile.isFile())) {
+        return true;
+      }
+      return failAndRunChecks(stdFile, altFile);
+    } else if (attRow.getAttType().equals(Config.ATT_TYPE_ZIP)) {
+      // need to check zip
+      if (stdFile.exists() && stdFile.isFile()) {
+        checkZip(stat, path.substring(0, path.length() - attName.length()), attName, attRow);
+        return true;
+      } else if (altFile.exists() && altFile.isFile()) {
+        checkZip(stat, altPath.substring(0, altPath.length() - attName.length()), attName, attRow);
+        return true;
+      }
+      return failAndRunChecks(stdFile, altFile);
+    }
+
+    // if we get to this point, it should be a normal file
+    if ((stdFile.exists() && stdFile.isFile()) || (altFile.exists() && altFile.isFile())) {
       return true;
-    } else {
-      return (new File(altPath)).exists();
+    }
+    return failAndRunChecks(stdFile, altFile);
+  }
+
+  private void logIfExistsAndDirOrFile(File f, String type, boolean shouldBeAFile) {
+    if (f.exists() && shouldBeAFile && f.isDirectory()) {
+      logger.warn("{} file location exists, but is a directory!  [{}]", type, f.getAbsolutePath());
+    } else if (f.exists() && !shouldBeAFile && f.isFile()) {
+      logger.warn("{} file location exists, but is a directory!  [{}]", type, f.getAbsolutePath());
+    }
+  }
+
+  private boolean failAndRunChecks(File stdFile, File altFile) {
+    logIfExistsAndDirOrFile(stdFile, "standard", true);
+    logIfExistsAndDirOrFile(altFile, "alternate", true);
+    return false;
+  }
+
+  private boolean failAndRunChecks(File stdFile, File altFile, boolean shouldBeAFile) {
+    logIfExistsAndDirOrFile(stdFile, "standard", shouldBeAFile);
+    logIfExistsAndDirOrFile(altFile, "alternate", shouldBeAFile);
+    return false;
+  }
+
+  private void checkZip(String stat, String path, String name, ResultsRow parentAttsRow) {
+    try {
+      List<ResultsRow> zipAtts = new ArrayList<>();
+      ZipFile zipFile = new ZipFile(path + Config.get(Config.GENERAL_OS_SLASH) + name);
+      // Unzipped files are stored in a directory named [zip-name].
+      // The zip attachment name's format is _zips/[zip-name].zip
+      final String unzippedLocation =
+          (path.length() > 7) ? name.substring(6) : "UNKNOWN_ZIP_LOCATION";
```

Comment:
zip paths have a hard limit of 6 characters?

Reply:
The initial path for storing the zips in the item's filestore directory always starts with a `_zips/`, so this check ensures that directory exists.